polynote-frontend/polynote/ui/component/notebook/cell.ts

```diff
@@ -613,21 +617,27 @@ class CodeCell extends Cell {
             }
         }
         cellState.state.presence.forEach(p => updatePresence(p.id, p.name, p.color, p.range))
-        cellState.view("presence").addObserver((newPresence, oldPresence) => {
-            const removed = diffArray(newPresence, oldPresence)[1]
+        cellState.observeKey("presence", (newPresence, update) => {
+            const removed = Object.values(update.removedValues ?? {})
 
             removed.forEach(p => {
-                const marker = presenceMarkers[p.id]
-                if (marker) {
-                    this.editor.deltaDecorations(marker, [])
+                if (p) {
+                    const marker = presenceMarkers[p.id]
+                    if (marker) {
+                        this.editor.deltaDecorations(marker, [])
+                    }
+                    delete presenceMarkers[p.id]
                 }
-                delete presenceMarkers[p.id]
             })
 
-            newPresence.forEach(p => {
-                updatePresence(p.id, p.name, p.color, p.range)
-            })
-        }, this)
+            const changed = update.changedValues(newPresence);
+            for (const key in changed) {
```

Comment:
If you use `for... of` you don't need to check `hasOwnProperty`

Reply:
`changedValues` is not an array, so you can't do `for... of`

Reply:
Ah, right, I was confused... `Object.keys(changed).forEach` would be similarly performant I think, and slightly cleaner to read imo. But, this is minor of course.